### PR TITLE
[WIP] feat: better template support

### PR DESCRIPTION
**Pull request by manniL**
Addresses issues left over in #202, especially that templates in the parent component are applied with child values.

Scenarios:
- [x] Parent template, child value
- [x] Parent template, child template, child value
- [x] Parent template, parent value child template (not sure if even possible, failing test is included)

### CODE REVIEW

**Comment by manniL** [comment]
cc @joshukraine, @moltar, @conceptdeluxe

File: test/getMetaInfo.spec.js [hunk 1]
```diff
@@ -363,4 +365,169 @@ describe('getMetaInfo', () => {
...
+      script: [],
+      noscript: [],
+      __dangerouslyDisableSanitizers: [],
+      __dangerouslyDisableSanitizersByTagID: {}
+    })
+  })
+
+  // TODO: Still failing :( Child template won't be applied if child has no content as well
+
+  it('properly uses meta templates with one-level-deep nested children template', () => {
```
Comment (manniL) [inline test/getMetaInfo.spec.js]: This one is still a problem :disappointed:

**Comment by joshukraine** [comment]
@manniL Thanks for working on this! 🙂

**Comment by Atinux** [comment]
Working on multi nested child with templates to make sure it works perfectly

**Comment by Atinux** [comment]
Well, at the moment it supports only one child with template, it's fine and more children should not happen in usage. Will see if people need this.

**Comment by conceptdeluxe** [comment]
Works like a charm :)

Thank you so much! 👍 💯

**Comment by Sylk** [comment]
Hey @manniL can I get a use case example of this? Because currently the documentation seems to reflect the older version of how to use this.

**Comment by manniL** [comment]
@Sylk Please check the tests for it for now. Don't have an example ready yet ☺️

**Comment by manniL** [comment]
@Sylk I've created the example from the tests
https://codesandbox.io/s/038mp9vkov

As you see, `App.vue` has an `og:title` set ("Test title") and also a template for it (```
template: chunk =>  `${chunk} - My page```).

The child component `HelloWorld.vue` has just a title set ("An important title!"). But the title (from HelloWorld) applied to the template (from App.vue) will be shown
image

**Pull request merged**